src/omnicore/omnicore.cpp

```diff
@@ -3048,6 +2888,54 @@ int mastercore::ClassAgnosticWalletTXBuilder(const std::string& senderAddress, c
     }
 }
 
+void CMPTxList::LoadActivations(int blockHeight)
+{
+    if (!pdb) return;
+
+    Slice skey, svalue;
+    Iterator* it = NewIterator();
+
+    PrintToLog("Loading feature activations from levelDB\n");
+
+    for(it->SeekToFirst(); it->Valid(); it->Next()) {
+       skey = it->key();
+       svalue = it->value();
+       string itData = svalue.ToString();
+
+       std::vector<std::string> vstr;
+       boost::split(vstr, itData, boost::is_any_of(":"), token_compress_on);
+       if (4 != vstr.size()) continue;
+       if (atoi(vstr[2]) != OMNICORE_MESSAGE_TYPE_ACTIVATION || atoi(vstr[0]) != 1) continue; // we only care about valid activations
+
+       uint256 hash(skey.ToString());
+       uint256 blockHash = 0;
+       CTransaction wtx;
+       CMPTransaction mp_obj;
+
+       if (!GetTransaction(hash, wtx, blockHash, true)) {
+           PrintToLog("ERROR: While loading activation transaction %s: tx in levelDB but does not exist.\n", hash.GetHex());
+           continue;
+       }
+       if (0 != ParseTransaction(wtx, blockHeight, 0, mp_obj)) {
```

Comment:
Nope :) but your point about the parsing rules is definitely prudent - though I prefer to pull height from the blockchain rather than levelDB (perhaps just habit hehe but I like going straight to the "primary source of truth") - how about something like this https://github.com/zathras-crypto/omnicore/commit/0b2ceddc78f2065a954f58928e9c541a35b375c7?